homeassistant/components/systemmonitor/sensor.py

```diff
@@ -309,6 +405,10 @@ def __init__(
         self._unique_id: str = slugify(f"{sensor_type}_{argument}")
         self._sensor_registry = sensor_registry
         self._argument: str = argument
+        self._attr_device_class: str | None = self.sensor_type[SENSOR_TYPE_DEVICE_CLASS]
+        self._attr_icon: str | None = self.sensor_type[SENSOR_TYPE_ICON]
+        self._attr_state_class: str | None = self.sensor_type[SENSOR_TYPE_STATE_CLASS]
+        self._attr_unit_of_measurement: str | None = self.sensor_type[SENSOR_TYPE_UOM]
```

Comment:
Instead of doing this, assign the entity description to the entity and this will work automatically.